src/pyodide/internal/loadPackage.ts

```diff
@@ -50,7 +60,23 @@ async function loadBundle(requirement: string): Promise<[string, ArrayBuffer]> {
   const decompressed = await decompressArrayBuffer(compressed);
 
   DiskCache.put(filename, compressed);
-  return [requirement, decompressed];
+  const reader = new ArrayBufferReader(decompressed);
+  return [requirement, reader];
+}
+
+async function loadBundleFromArtifactBundler(
+  requirement: string
+): Promise<[string, Reader]> {
+  const packagesVersion = PACKAGES_VERSION;
+  const filename = LOCKFILE['packages'][requirement]['file_name'];
+  const fullPath = 'python-package-bucket/' + packagesVersion + '/' + filename;
+  const reader = ArtifactBundler.getPackage(fullPath);
+  if (!reader)
+    throw new Error(
+      'Failed to get package ' + fullPath + ' from ArtifactBundler'
+    );
+  return Promise.resolve([requirement, reader]);
+  // ^ this is okay to do during startup since it resolves immediately
```

Comment:
This is weird, presumably it would work to directly return the value right?
```suggestion
  return [requirement, reader];
```
(We don't even necessarily need to make the function async, if you do `await notThenable` it just returns `notThenable`. But this is perhaps not ideal for readability.)